### PR TITLE
[Feature][Paimon] Support specify paimon table write properties, partition keys and primary keys

### PR DESCRIPTION
**Pull request by dailai**
### Purpose of this pull request

[Feature][Paimon] Support specify paimon table write properties, partition keys and primary keys
[Feature][Paimon] Support timestamp(n) for the paimon sink
[Feature][API] ColumnParser support define columnScale which is useful for fakesource

For example:
1. file.format of paimon properties support orc, parquet, avro
2. bucket of paimon properties support [Normal Dynamic Bucket Mode](https://paimon.apache.org/docs/0.6/concepts/primary-key-table/#normal-dynamic-bucket-mode)
3. paimon.table.partition-keys can support partition table of paimon
4. Solve the timestamp(n) problem for the paimon sink
5. Schema.columns can define columnScale for fakesource

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  6. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  7. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/utils/SchemaUtil.java [hunk 1]
```diff
@@ -35,20 +37,39 @@ public static DataType toPaimonType(Column column) {
...
             paiSchemaBuilder.column(column.getName(), toPaimonType(column));
         }
-        PrimaryKey primaryKey = tableSchema.getPrimaryKey();
-        if (Objects.nonNull(primaryKey) && primaryKey.getColumnNames().size() > 0) {
-            paiSchemaBuilder.primaryKey(primaryKey.getColumnNames());
+        List<String> primaryKeys = paimonSinkConfig.getPrimaryKeys();
+        if (primaryKeys.isEmpty() && Objects.nonNull(tableSchema.getPrimaryKey())) {
+            primaryKeys = tableSchema.getPrimaryKey().getColumnNames();
+        }
+        if (Objects.nonNull(primaryKeys) && primaryKeys.size() > 0) {
```
Comment (Carl-Zhou-CN) [inline seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/utils/SchemaUtil.java]: It seems impossible to be null

Comment (dailai) [reply seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/utils/SchemaUtil.java]: Yes, I've removed it.

**Review by Carl-Zhou-CN — APPROVED** [review]
LGTM

**Comment by Hisoka-X** [comment]
cc @TyrantLucifer

**Review by hailin0 — APPROVED** [review]
LGTM

**Pull request merged**